packages/worker/src/worker.ts

```diff
@@ -764,10 +811,18 @@ export class Worker {
           return { activation, parentSpan };
         });
       } catch (err) {
-        parentSpan.setStatus({ code: otel.SpanStatusCode.ERROR }).end();
+        if (!(err instanceof errors.ShutdownError)) {
+          parentSpan.setStatus({ code: otel.SpanStatusCode.ERROR }).end();
+        }
         throw err;
       }
-    });
+    }).pipe(
+      tap({
+        complete: () => {
+          this.workflowPollerStateSubject.next('SHUTDOWN');
```

Comment:
We should change the state to `FAILED` on error here and perhaps even flush the cache when that state is reached.

Reply:
Seems like just falling through to https://github.com/temporalio/sdk-typescript/blob/ce23c2c96b3e9bbaa46378d5658f6698d3f6bd5f/packages/worker/src/worker.ts#L996 is the right place to handle that? I think cache flushing subject to that probably goes in another PR.